src/upload/UploadPage.jsx

```diff
@@ -23,16 +40,71 @@ class UploadPage extends React.Component {
               Please reach out to <a href="mailto:partner-support@edx.org">partner-support@edx.org</a> requesting access to the Registrar service.
             </p>
           )}
-          open={this.props.data.length === 0}
+          open={!this.props.authorized}
         />
         {this.props.data.length > 0 && this.props.data.map(program => (
-          <div className="container" key={program.program_key}>
-            <h2>{program.program_title}</h2>
+          <div className="container" key={program.programKey}>
+            <h2>{program.programTitle}</h2>
+            {this.props.programBanners[program.programKey] &&
+              !!this.props.programBanners[program.programKey].length &&
+              this.props.programBanners[program.programKey].map(banner => (
+                <StatusAlert
+                  dismissible
+                  open
+                  key={banner.id}
+                  alertType={banner.bannerType}
+                  onClose={() => this.props.removeBanner(program.programKey, banner.id)}
+                  dialog={(
+                    <div className="modal-alert">
+                      {`${banner.message} `}
+                      {banner.linkMessage && <a href={banner.linkHref} target="_blank" rel="noopener noreferrer">{banner.linkMessage}</a>}
+                    </div>
+                  )}
+                />
+            ))}
+            <div className="btn-group" role="group">
+              <button className="btn btn-outline-primary">
+                <input
+                  type="file"
+                  visability="hidden"
+                  style={{
+                    position: 'absolute',
+                    height: '100%',
+                    width: '100%',
+                    opacity: '0',
+                    top: '0',
+                    left: '0',
+                  }}
+                  onChange={e => this.handleUploadProgramEnrollments(program.programKey, e)}
+                />Upload Program Enrollments
+              </button>
+              <button
+                className="btn btn-outline-primary"
+                onClick={() => this.handleDownloadProgramEnrollments(program.programKey)}
+              >Download Program Enrollments
+              </button>
+            </div>
             <div className="btn-group" role="group">
-              <button className="btn btn-outline-primary">Upload Program Enrollments</button>
-              <button className="btn btn-outline-primary">Upload Course Enrollments</button>
-              <button className="btn btn-outline-primary">Download Program Enrollments</button>
-              <button className="btn btn-outline-primary">Download Course Enrollments</button>
+              <button className="btn btn-outline-primary">
```

Comment:
#NIT should we consider moving the upload button into it's own component file? The styles are repeated twice.

Reply:
Eventually, I think there is room to refactor this but would be hesitant to do so as of now without knowing how Program Manager may evolve in future iterations